### PR TITLE
Update __init__.py to use a base_url valid for both single and multiuser server

### PR DESCRIPTION
**Pull request by epifanio**
fix the base_url path so to be able to load the jupyterlab  in a jupyterhub multiuser environment. I tested few functionalities new notebooks, terminal and new text file are working properly.

I installed the jupyterlub system wide with:

```sudo python3 setup.py install```

and enabled the extension for each user logging in with a fresh ssh session and then issuing the command:
```jupyter serverextension enable --py jupyterlab```

Thanks @jcb91 for suggesting the needed changes!

### CODE REVIEW

File: jupyterlab/__init__.py [hunk 1]
```diff
@@ -45,8 +45,9 @@ def _jupyter_server_extension_paths():
         "module": "jupyterlab"
     }]
     
+from notebook.utils import url_path_join as ujoin
```
Comment (jcb91) [inline jupyterlab/__init__.py]: Module-level imports are best at the top of the file, to be less confusing later

**Comment by jcb91** [comment]
Looks good, though I think as @jasongrout mentioned somewhere, it might be that the mathjax URL needs the base_url adding also.

File: jupyterlab/__init__.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ class LabHandler(IPythonHandler):
     @web.authenticated
     def get(self):
         self.write(self.render_template('lab.html',
-            static_prefix=PREFIX,
+            static_prefix=ujoin(self.application.settings['base_url'], '/lab'),
```
Comment (jasongrout) [inline jupyterlab/__init__.py]: Can you use PREFIX instead of /lab here?

File: jupyterlab/__init__.py [hunk 3]
```diff
@@ -45,8 +45,9 @@ def _jupyter_server_extension_paths():
...
     }]
     
+from notebook.utils import url_path_join as ujoin
 def load_jupyter_server_extension(nbapp):
     nbapp.log.info('Pre-alpha version of JupyterLab extension loaded from %s'%HERE)
     webapp = nbapp.web_app
-    #base_url = webapp.settings['base_url']
-    webapp.add_handlers(".*$", default_handlers)
+    base_url = webapp.settings['base_url']
+    webapp.add_handlers(".*$", [(ujoin(base_url, h[0]),) + h[1:] for h in default_handlers])
```
Comment (jasongrout) [inline jupyterlab/__init__.py]: This should probably use the OS path join here since this is a filesystem path.

Comment (jcb91) [inline jupyterlab/__init__.py]: Pretty sure it's a URI regex for the handler to handle: see similar operation at https://github.com/jupyter/notebook/blob/45e16b22788a3de401e6ba204c256d8a381156bd/notebook/notebookapp.py#L316-322

**Comment by SylvainCorlay** [comment]
@epifanio thanks! it seems that you opened this PR to your own fork.

File: jupyterlab/__init__.py (hunk 2 above)
Comment (SylvainCorlay) [inline jupyterlab/__init__.py]: I am not sure we need `lab` anymore since we merged my PR which forwards the to lab.

**Comment by epifanio** [comment]
@SylvainCorlay should I make a patch against git master?

File: jupyterlab/__init__.py (hunk 3 above)
Comment (epifanio) [inline jupyterlab/__init__.py]: How should I fix the mathjax URL prefix? will be something like:

```
- mathjax_url=self.mathjax_url,
+ static_prefix=ujoin(self.application.settings['base_url'], self.mathjax_url),
```
be enough?

**Comment by SylvainCorlay** [comment]
@epifanio just use gihub interface to open a PR from the same branch to the main repo.

**Comment by epifanio** [comment]
I opened it here https://github.com/jupyter/jupyterlab/pull/40  i'll close this

**Pull request closed without merging**